### PR TITLE
make it work on macOS

### PR DESCRIPTION
**Pull request by erikbor**
@phuonghuynh I added support for macOS, can you please verify that my changes do not impact running on Linux?

Thanks

### CODE REVIEW

File: scripts/__utils__.sh [hunk 1]
```diff
@@ -5,8 +5,14 @@ if [ "${__utils__}" = true ] ; then
...
-  local output=$(printf "$1" | openssl dgst -sha256)
-  printf "${output}"
+  uname | grep -q ^Darwin
+  if [ $? -eq 0 ]; then
+    local output="$(printf "$1" | ${OPENSSL_BIN} dgst -sha256 -binary | xxd -p -c 256)"
+  else
+    local output="$(printf "$1" | ${OPENSSL_BIN} dgst -sha256)"
+  fi
+
+  echo "${output}"
```
Comment (phuonghuynh) [inline scripts/__utils__.sh]: Could we use "printf"? since "echo" return new-line-char which might made some unexpected errors in some env

**Comment by phuonghuynh** [comment]
Could we reformat the code as defined in [.editorconfig](https://github.com/riboseinc/aws-authenticating-secgroup-scripts/blob/master/.editorconfig) ?

[editorconfig](http://editorconfig.org/) has some plugin for some IDEs like vim, IntelliJ...

**Comment by ronaldtse** [comment]
@phuonghyunh good call. Could you actually directly edit this PR to fix these? Thanks!

File: scripts/__os_openssl__.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		echo "Homebrew is a prerequisite to install the latest OpenSSL on macOS (hint: install brew via 'https://brew.sh/')" >&2
+		exit 1
+	fi	
+
+	OPENSSL_BIN="$(brew list openssl | awk '/\/bin\/openssl/')"
+	if [[ -z "${OPENSSL_BIN}" ]]; then
+		echo "Homebrew OpenSSL not installed (hint: 'brew install openssl')" >&2
+		exit 1
+	fi
+fi
```
Comment (phuonghuynh) [inline scripts/__os_openssl__.sh]: we could reformat this as discussed at https://github.com/riboseinc/aws-authenticating-secgroup-scripts/pull/6#issuecomment-355720664

File: scripts/__utils__.sh [hunk 3]
```diff
@@ -18,7 +24,14 @@ function log() {
...
+  if [ $? -eq 0 ]; then
+    local hash="$(printf "$2" | ${OPENSSL_BIN} dgst -binary -sha256 -mac HMAC -macopt hexkey:"$1" | xxd -p -c 256)"
+  else
+    local hash="$(printf "$2" | openssl dgst -binary -hex -sha256 -mac HMAC -macopt hexkey:"$1")"
+  fi
+
+  echo "${hash}"
 }
 
 __utils__=true
```
Comment (phuonghuynh) [inline scripts/__utils__.sh]: we could reformat this as discussed at https://github.com/riboseinc/aws-authenticating-secgroup-scripts/pull/6#issuecomment-355720664

**Comment by phuonghuynh** [comment]
Yes, I agree but i not test for RHEL & CentOS

**Comment by phuonghuynh** [comment]
@erikbor @ronaldtse I updated this PR, tested on both MacOS 10.13.2 & Ubuntu 16.04.1 LTS
 - Fix for Ubuntu 16
 - Remove command "xxd"

Could we update the OS list in #7 ?

**Review by ronaldtse — APPROVED** [review]
LGTM

**Comment by ronaldtse** [comment]
Thanks @phuonghuynh !

**Pull request merged**